This patch removes a dummy check on proxy usage before applying authentication.
This was caused by a copy-paste error in older revision of this file, and then
drifted to the current problem while the code evolved.  With this simple
change, authenticated streams works just fine.

A big thanks to Mark Kretschmann from the amaroK team who noticed the problem
and helped tracking it down.

Index: xine-lib-1.1.2cvs20060328/src/input/input_http.c
===================================================================
--- xine-lib-1.1.2cvs20060328.orig/src/input/input_http.c
+++ xine-lib-1.1.2cvs20060328/src/input/input_http.c
@@ -657,7 +657,7 @@ static int http_plugin_open (input_plugi
   if (this->port == 0)
     this->port = DEFAULT_HTTP_PORT;
   
-  if ((this->user && strlen(this->user)) && use_proxy) {
+  if (this->user && strlen(this->user)) {
     if (http_plugin_basicauth (this->user, this->password, this->auth, BUFSIZE)) {
       _x_message(this->stream, XINE_MSG_CONNECTION_REFUSED, "basic auth error", NULL);
       return 0;
